tools/tertiary-analysis/decoupler/decoupler_pathway_inference.xml

```diff
@@ -0,0 +1,84 @@
+<tool id="decoupler_pathway_inference" name="Decoupler Pathway Inference" version="1.4.0+galaxy0" profile="20.05" license="MIT">
+    <description>
+        Pathway inference using the Decoupler.
+    </description>
+    <requirements>
+        <requirement type="package" version="1.4.0">decoupler</requirement>
+    </requirements>
+    <command>
+        python '$__tool_directory__/decoupler_pathway_inference.py'
+            -i '$input_anndata'
+            -n '$input_network_file'
+            --min_n "$min_n"
+            $use_raw
+            --output "inference"
+            $write_activities_path
+    </command>
+    <inputs>
+        <param name="input_anndata" type="data" format="h5ad" label="Input AnnData file" />
+        <param name="input_network_file" type="data" format="tabular" label="Input Network file" />
+        <param name="min_n" type="integer" min="0" value="5" label="Minimum of targets per source. If less, sources are removed" />
+        <param name="use_raw" type="boolean" truevalue="--use_raw" falsevalue="" checked="false" label="Whether to use the raw part of the AnnData object" />
+        <param name="write_activities_path" type="boolean" truevalue="--activities_path anndata_activities_path.h5ad" falsevalue="" checked="true" label="Write the modified AnnData object" />
+    </inputs>
+    <outputs>
+        <data name="output_ad" format="h5ad" from_work_dir="anndata_activities_path.h5ad" label="${tool.name} on ${on_string}: Output AnnData file">
+            <filter>write_activities_path</filter>
+        </data>
+        <data name="output_table_mlm" format="tabular" from_work_dir="inference_mlm.tsv" label="${tool.name} on ${on_string}: Output MLM estimate table" />
+    </outputs>
+    <tests>
+        <!-- Hint: You can use [ctrl+alt+t] after defining the inputs/outputs to auto-scaffold some basic test cases. -->
+
+    <test expect_num_outputs="2">
+        <param name="input_anndata" value="pbmc3k_processed.h5ad"/>
+        <param name="input_network_file" value="progeny_test.tsv"/>
+        <param name="min_n" value="0"/>
+        <param name="use_raw" value="false"/>
+        <param name="write_activities_path" value="true"/>
+        <output name="output_ad">
+            <assert_contents>
+                <has_h5_keys keys="obsm/mlm_estimate"/>
+            </assert_contents>
+        </output>
+        <output name="output_table_mlm">
+            <assert_contents>
+                <has_n_columns n="5"/>
+            </assert_contents>
+        </output>
+    </test>
+    </tests>
+    <help>
+**What it does**
+
+Usage
+.....
+
+
+**Description**
+
+This tool scores cells using the AUCell method for gene sets.
```

Comment:
This is not using AUCell, but MLM. Ideally it should also support ULM I think if the method call and output format are the same.

Reply:
this is now fixed, ULM added as well